src/main/java/com/ericsson/eiffel/remrem/publish/service/MessageServiceRMQImpl.java

```diff
@@ -1,107 +1,167 @@
+            if (eventId != null) {
+                map.put(eventId, json.toString());
+            } else {
+                List<PublishResultItem> events = new ArrayList<>();
+                createFailureResult(events);
+                return new SendResult(events);
+            }
+            return send(routingKey, map);
+        } catch (final JsonSyntaxException e) {
             String resultMsg = "Could not parse JSON.";
             if (e.getCause() != null) {
                 resultMsg = resultMsg + " Cause: " + e.getCause().getMessage();
             }
             log.error(resultMsg, e.getMessage());
-            List<SendResult> results = new ArrayList<>();
-            results.add(new SendResult(resultMsg));
-            return results;
-    	}
+            List<PublishResultItem> events = new ArrayList<>();
+            createFailureResult(events);
+            return new SendResult(events);
+        }
     }
-    
-    /* (non-Javadoc)
-     * @see com.ericsson.eiffel.remrem.publish.service.MessageService#send(java.lang.String, com.google.gson.JsonElement)
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * com.ericsson.eiffel.remrem.publish.service.MessageService#send(java.lang.
+     * String, com.google.gson.JsonElement)
      */
-    @Override public List<SendResult> send(String routingKey, JsonElement json) {
-    	List<String> msgs = new ArrayList<>();
-    	
-    	if (json == null) {    		
-            String resultMsg = "Invalid json content.";
-            log.error(resultMsg);
-            List<SendResult> results = new ArrayList<>();
-            results.add(new SendResult(resultMsg));
-            return results;
-    	}
+    @Override
+    public SendResult send(String routingKey, JsonElement json) {
+        Map<String, String> map = new HashMap<>();
+        List<PublishResultItem> events = new ArrayList<>();
+        if (json == null) {
+            createFailureResult(events);
+        }
+        if (json.isJsonArray()) {
+            JsonArray bodyJson = json.getAsJsonArray();
+            for (JsonElement obj : bodyJson) {
+                String eventId = rmqHelper.getEventId(obj);
+                if (eventId != null) {
+                    map.put(eventId, obj.toString());
+                } else {
+                    createFailureResult(events);
+                }
+            }
+        } else {
+            String eventId = rmqHelper.getEventId(json);
+            if (eventId != null) {
+                map.put(eventId, json.toString());
+            } else {
+                createFailureResult(events);
+            }
+        }
+
+        if (map.size() > 0) {
+            SendResult result = send(routingKey, map);
+            events.addAll(result.getEvents());
+            result.setEvents(events);
+            return result;
+        } else {
+            SendResult result = new SendResult();
+            result.setEvents(events);
+            return result;
+        }
```

Comment:
I think if else above is easier to read if written  like below:
SendResult result = new SendResult();
 if (map.size() > 0) {
            result = send(routingKey, map);
            events.addAll(result.getEvents());
        } 
           
            result.setEvents(events);
            return result;